contribs/gmf/src/directives/drawfeature.js

```diff
@@ -0,0 +1,399 @@
+  ol.events.listen(drawRectangle,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // === TEXT ===
+
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.drawText = new ol.interaction.Draw({
+    type: ol.geom.GeometryType.POINT
+  });
+
+  var drawText = this.drawText;
+  this.interactions_.push(drawText);
+  drawText.setActive(false);
+  ngeoDecorateInteraction(drawText);
+  this.map.addInteraction(drawText);
+  ol.events.listen(drawText, ol.interaction.DrawEventType.DRAWEND,
+      this.handleDrawEnd_.bind(this, geomType.TEXT), this);
+  ol.events.listen(drawText,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      this.handleActiveChange_, this);
+
+
+  // Watch the "active" property, and disable the draw interactions
+  // when "active" gets set to false.
+  $scope.$watch(
+    function() {
+      return this.active;
+    }.bind(this),
+    function(newVal) {
+      if (newVal === false) {
+        this.interactions_.forEach(function(interaction) {
+          interaction.setActive(false);
+        }, this);
+      }
+    }.bind(this)
+  );
+
+};
+
+
+/**
+ * @param {ol.ObjectEvent} event Event.
+ * @private
+ */
+gmf.DrawfeatureController.prototype.handleActiveChange_ = function(event) {
+  var active;
+  this.interactions_.some(function(interaction) {
+    active = interaction.getActive();
+    return active;
+  }, this);
+  this.active = active;
+};
+
+
+/**
+ * Called when a feature is finished being drawn. Set the default properties
+ * for its style, then set its style and add it to the features collection.
+ * @param {string} type Type of geometry being drawn.
+ * @param {ol.interaction.DrawEvent|ngeo.MeasureEvent} event Event.
+ * @private
+ */
+gmf.DrawfeatureController.prototype.handleDrawEnd_ = function(type, event) {
+  var feature = event.feature;
+
+  var prop = ngeo.FeatureProperties;
+
+  /**
+   * @type {string}
+   */
+  var name;
+  switch (type) {
+    case ngeo.GeometryType.CIRCLE:
+      name = this.gettextCatalog_.getString('Circle');
+
+      var featureGeom = /** @type {ol.geom.Circle} */ (feature.getGeometry());
+      feature.setGeometry(
+          ol.geom.Polygon.fromCircle(featureGeom, 64)
+      );
+      feature.set(prop.IS_CIRCLE, true);
+      break;
+    case ngeo.GeometryType.POINT:
+      name = this.gettextCatalog_.getString('Point');
+      break;
+    case 'text':
+      name = this.gettextCatalog_.getString('Text');
+      feature.set(prop.IS_TEXT, true);
+      break;
+    case ngeo.GeometryType.LINESTRING:
+      name = this.gettextCatalog_.getString('LineString');
+      break;
+    case ngeo.GeometryType.POLYGON:
+      name = this.gettextCatalog_.getString('Polygon');
+      break;
+    case ngeo.GeometryType.RECTANGLE:
+      name = this.gettextCatalog_.getString('Rectangle');
+      feature.set(prop.IS_RECTANGLE, true);
+      break;
+    default:
```

Comment:
name = this.gettextCatalog_.getString('LineString');

This line could go in default ? As you are doing it for all